crates/eframe/src/native/epi_integration.rs

```diff
@@ -426,6 +432,12 @@ impl EpiIntegration {
             WindowEvent::ScaleFactorChanged { scale_factor, .. } => {
                 self.frame.info.native_pixels_per_point = Some(*scale_factor as _);
             }
+            #[cfg(feature = "dark-light")]
+            WindowEvent::ThemeChanged(winit_theme) if self.follow_system_theme => {
```

Comment:
I don't see any reason to limit this on the `dark-light` feature.

`dark-light` pulls in a specific dependency for detecting the system theme on start-up. Maybe we can do that via `winit` instead?

Reply:
I have removed `#[cfg(feature = "dark-light")]` from all places where it's not necessary.

Detecting the theme via winit's `theme` function works for Windows and macOS—I have tested the latest change on both my Windows 11 and my macOS devices.

The `theme` function always returns `None` on Linux though (for both [X11](https://github.com/rust-windowing/winit/blob/08bdca19b14fae099076e51816c799d4902fdcbb/src/platform_impl/linux/x11/window.rs#L1620) and [Wayland](https://github.com/rust-windowing/winit/blob/08bdca19b14fae099076e51816c799d4902fdcbb/src/platform_impl/linux/wayland/window/mod.rs#L689)), so there we still have to rely on `dark-light`.

The `NativeOptions::system_theme` function has become a bit of a liar, since it no longer corresponds with the effectively used system theme on Windows and macOS.